chat-server/index.html

```diff
@@ -1,24 +1,123 @@
 <!DOCTYPE html>
 <html lang="en">
-  <head>
-    <title>Welcome to CYF Chat</title>
-  </head>
-  <body>
-    <header><h1>CYF Chat</h1></header>
-    <main>
+
+<head>
+  <meta charset="UTF-8">
+  <meta name="viewport" content="width=device-width, initial-scale=1.0">
+  <link rel="stylesheet" href="style.css">
+  <title>Welcome to CYF Chat</title>
+</head>
+
+<body>
+  <header>
+    <h1>CYF Chat</h1>
+  </header>
+  <main>
+    <section id="send-message">
       <h2>Send a message</h2>
-      <form action="/messages" method="post">
-        <label
-          >Name: <input type="text" name="from" placeholder="Your Name"
-        /></label>
-        <label
-          >Message: <input type="text" name="text" placeholder="The message..."
-        /></label>
+      <form id="message-form" action="/messages" method="post">
+        <label>Name: <input id="new-message-from" type="text" name="from" placeholder="Your Name" /></label>
+        <label>Message: <input id="new-message-text" type="text" name="text" placeholder="The message..." /></label>
         <button type="submit">Send</button>
       </form>
-    </main>
-    <footer>
-      <h3><a href="/messages">See all messages</a></h3>
-    </footer>
-  </body>
-</html>
+    </section>
+    <section id="messages">
+      <h3><a href="" onclick="fetchAndLoadMessages()">See all messages</a></h3>
+      <div>
+        <h3>See message by ID</h3>
+        <input id="find-message-id" type="text" placeholder="Type an ID here">
+        <button onclick="fetchAndLoadMessageById()">Show message</button>
+      </div>
+      <ul id="messages-list">
+
+      </ul>
+    </section>
+  </main>
+  <footer>
+    <!-- <p>Project by Pedro Ricciardi, as part of the CYF Course - Servers Module - Chat-Server Issue</p> -->
+  </footer>
+
+  <!-- SCRIPTS -->
+  <script>
+    const renderMessage = (message) => {
```

Comment:
How do you feel about using the DOM APIs directly here vs using something like a `<template>` tag? What are the advantages/disadvantages of both?

Reply:
Usually, I prefer to use `template` because it allows me to create the page, visualise the layout, and ensure that everything appears as expected. However, for simpler tasks, especially when I'm short on time and need to create something quickly, I prefer to create the tags directly using createElement.